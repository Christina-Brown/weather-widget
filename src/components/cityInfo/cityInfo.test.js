import React from "react";
import { shallow } from "enzyme";
import CityInfo from "./cityInfo";

let wrapper;

beforeEach(() => {
  const mockProps = {
    coord: {
      lat: 51.4237,
      lon: -0.36,
    },
    dt: 1613327589,
    id: 2647550,
    main: {
      temp: 3.01,
    },
    name: "Hampton",
    sys: {
      country: "GB",
      id: 1417,
      sunrise: 1613287023,
      sunset: 1613322865,
    },
    weather: [
      {
        description: "broken clouds",
        icon: "04n",
        id: 803,
        main: "Clouds",
      },
    ],
  };

  wrapper = shallow(<CityInfo weatherData={mockProps} />);
});

it("should render CityInfo without crashing", () => {
  expect(wrapper).toMatchSnapshot();
});

it("should call correct image source for icon ", () => {
  const Img = wrapper.find("img");
  expect(Img.getElement(0).props.src).toEqual(
    "http://openweathermap.org/img/wn/04n@2x.png"
  );
});

it("should show only one image ", () => {
  const Img = wrapper.find("img");
  expect(Img.length).toEqual(1);
});
