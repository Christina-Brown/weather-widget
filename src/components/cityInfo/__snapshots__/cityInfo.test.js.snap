// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CityInfo without crashing 1`] = `
<div
  className="city-info"
>
  <div
    className="location"
  >
    Hampton
    , 
    GB
  </div>
  <div
    className="date"
  >
    Monday 15 February 2021
  </div>
  <div
    className="weather-info"
  >
    <div
      className="temp"
    >
      3
      Â°C
    </div>
    <div
      className="weather"
    >
      <img
        className="weather"
        src="http://openweathermap.org/img/wn/04n@2x.png"
      />
      Clouds
    </div>
  </div>
</div>
`;
